NDTensors/ext/NDTensorsMetalExt/mul.jl

```diff
@@ -10,3 +10,13 @@ function LinearAlgebra.mul!(
   mul!(transpose(CM), transpose(BM), transpose(AM), α, β)
   return unexpose(CM)
 end
+
+# This was calling generic matrix multiplication.
+# TODO: Raise an issue with `Metal.jl`.
+function LinearAlgebra.mul!(
+  CM::Exposed{<:MtlArray,<:Adjoint}, AM::Exposed{<:MtlArray}, BM::Exposed{<:MtlArray}, α, β
+)
+  return mul!(parent(CM), BM', AM', α, β)
+
```

Comment:
```suggestion
  mul!(CM', BM', AM', α, β)
```

Reply:
made these changes to match (Sorry was using copy paste and missed the `return mul!` portion. All are fixed now